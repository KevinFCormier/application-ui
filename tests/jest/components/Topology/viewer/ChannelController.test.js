/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
"use strict";

import React from "react";
import { shallow } from "enzyme";
import { mount } from "enzyme";

import ChannelController from "../../../../../src-web/components/Topology/viewer/ChannelControl";

const channelController = {
  activeChannel: "__ALL__/__ALL__//__ALL__/__ALL__",
  isChangingChannel: undefined,
  changeTheChannel: jest.fn,
  allChannels: [
    "default/guestbook-app//gbapp-ch/guestbook-app-latest///cassandra-cassandra-service///mysql-wordpress-pd-wordpress-mysql-deployment",
    "default/guestbook-app//gbapp-ch/guestbook-app-latest///mysql-wordpress-pd-wordpress-mysql-service///staging-elasticsearch-elasticsearch-serviceaccount"
  ]
};
describe("EditorBar components", () => {
  it("empty", () => {
    const component = shallow(<ChannelController />);
    expect(component).toMatchSnapshot();
  });

  it("default", () => {
    const component = shallow(
      <ChannelController channelControl={channelController} locale={"en-US"} />
    );
    expect(component).toMatchSnapshot();
  });
});

describe("EditorBar components", () => {
  it("empty", () => {
    const component = shallow(<ChannelController />);
    expect(component).toMatchSnapshot();
  });

  it("default", () => {
    const wrapper = mount(
      <ChannelController channelControl={channelController} locale={"en-US"} />
    );

    wrapper
      .find(".channel-control-subchannel")
      .at(0)
      .simulate("click");

    wrapper
      .find(".channel-control-subchannel")
      .at(0)
      .simulate("keypress");
  });
});
