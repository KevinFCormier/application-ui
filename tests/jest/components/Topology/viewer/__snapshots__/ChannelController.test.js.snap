// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorBar components default 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChannelControl
    channelControl={
      Object {
        "activeChannel": "__ALL__/__ALL__//__ALL__/__ALL__",
        "allChannels": Array [
          "default/guestbook-app//gbapp-ch/guestbook-app-latest///cassandra-cassandra-service///mysql-wordpress-pd-wordpress-mysql-deployment",
          "default/guestbook-app//gbapp-ch/guestbook-app-latest///mysql-wordpress-pd-wordpress-mysql-service///staging-elasticsearch-elasticsearch-serviceaccount",
        ],
        "changeTheChannel": [Function],
        "isChangingChannel": undefined,
      }
    }
    locale="en-US"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoHeight": false,
      "autoHeightMax": 200,
      "autoHeightMin": 0,
      "autoHide": false,
      "autoHideDuration": 200,
      "autoHideTimeout": 1000,
      "children": <div
        className="channel-controls-container"
      >
        <div
          aria-label="0"
          className="channel-control"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        />
        <div
          aria-label="1"
          className="channel-control channel-control-subchannel selected"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <React.Fragment>
            <div
              className="channel-control-start"
            >
              1
            </div>
          </React.Fragment>
        </div>
        <div
          aria-label="2"
          className="channel-control channel-control-subchannel"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <React.Fragment>
            <div
              className="channel-control-start"
            >
              2
            </div>
          </React.Fragment>
        </div>
      </div>,
      "className": "channel-controls-container",
      "hideTracksWhenNotNeeded": false,
      "renderThumbHorizontal": [Function],
      "renderThumbVertical": [Function],
      "renderTrackHorizontal": [Function],
      "renderTrackVertical": [Function],
      "renderView": [Function],
      "style": Object {
        "height": 0,
        "width": 20,
      },
      "tagName": "div",
      "thumbMinSize": 30,
      "universal": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            aria-label="0"
            className="channel-control"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          />,
          <div
            aria-label="1"
            className="channel-control channel-control-subchannel selected"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            <React.Fragment>
              <div
                className="channel-control-start"
              >
                1
              </div>
            </React.Fragment>
          </div>,
          <div
            aria-label="2"
            className="channel-control channel-control-subchannel"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            <React.Fragment>
              <div
                className="channel-control-start"
              >
                2
              </div>
            </React.Fragment>
          </div>,
        ],
        "className": "channel-controls-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "default/guestbook-app//gbapp-ch/guestbook-app-latest,default,guestbook-app,gbapp-ch,guestbook-app-latest",
          "nodeType": "host",
          "props": Object {
            "aria-label": "0",
            "children": undefined,
            "className": "channel-control",
            "onClick": [Function],
            "onKeyPress": [Function],
            "role": "button",
            "tabIndex": "0",
          },
          "ref": [Function],
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "default/guestbook-app//gbapp-ch/guestbook-app-latest///cassandra-cassandra-service///mysql-wordpress-pd-wordpress-mysql-deployment",
          "nodeType": "host",
          "props": Object {
            "aria-label": "1",
            "children": <React.Fragment>
              <div
                className="channel-control-start"
              >
                1
              </div>
            </React.Fragment>,
            "className": "channel-control channel-control-subchannel selected",
            "onClick": [Function],
            "onKeyPress": [Function],
            "role": "button",
            "tabIndex": "0",
          },
          "ref": [Function],
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <div
                className="channel-control-start"
              >
                1
              </div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "className": "channel-control-start",
              },
              "ref": null,
              "rendered": 1,
              "type": "div",
            },
            "type": Symbol(react.fragment),
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "default/guestbook-app//gbapp-ch/guestbook-app-latest///mysql-wordpress-pd-wordpress-mysql-service///staging-elasticsearch-elasticsearch-serviceaccount",
          "nodeType": "host",
          "props": Object {
            "aria-label": "2",
            "children": <React.Fragment>
              <div
                className="channel-control-start"
              >
                2
              </div>
            </React.Fragment>,
            "className": "channel-control channel-control-subchannel",
            "onClick": [Function],
            "onKeyPress": [Function],
            "role": "button",
            "tabIndex": "0",
          },
          "ref": [Function],
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <div
                className="channel-control-start"
              >
                2
              </div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2,
                "className": "channel-control-start",
              },
              "ref": null,
              "rendered": 2,
              "type": "div",
            },
            "type": Symbol(react.fragment),
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoHeight": false,
        "autoHeightMax": 200,
        "autoHeightMin": 0,
        "autoHide": false,
        "autoHideDuration": 200,
        "autoHideTimeout": 1000,
        "children": <div
          className="channel-controls-container"
        >
          <div
            aria-label="0"
            className="channel-control"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          />
          <div
            aria-label="1"
            className="channel-control channel-control-subchannel selected"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            <React.Fragment>
              <div
                className="channel-control-start"
              >
                1
              </div>
            </React.Fragment>
          </div>
          <div
            aria-label="2"
            className="channel-control channel-control-subchannel"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
          >
            <React.Fragment>
              <div
                className="channel-control-start"
              >
                2
              </div>
            </React.Fragment>
          </div>
        </div>,
        "className": "channel-controls-container",
        "hideTracksWhenNotNeeded": false,
        "renderThumbHorizontal": [Function],
        "renderThumbVertical": [Function],
        "renderTrackHorizontal": [Function],
        "renderTrackVertical": [Function],
        "renderView": [Function],
        "style": Object {
          "height": 0,
          "width": 20,
        },
        "tagName": "div",
        "thumbMinSize": 30,
        "universal": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              aria-label="0"
              className="channel-control"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
            />,
            <div
              aria-label="1"
              className="channel-control channel-control-subchannel selected"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
            >
              <React.Fragment>
                <div
                  className="channel-control-start"
                >
                  1
                </div>
              </React.Fragment>
            </div>,
            <div
              aria-label="2"
              className="channel-control channel-control-subchannel"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex="0"
            >
              <React.Fragment>
                <div
                  className="channel-control-start"
                >
                  2
                </div>
              </React.Fragment>
            </div>,
          ],
          "className": "channel-controls-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "default/guestbook-app//gbapp-ch/guestbook-app-latest,default,guestbook-app,gbapp-ch,guestbook-app-latest",
            "nodeType": "host",
            "props": Object {
              "aria-label": "0",
              "children": undefined,
              "className": "channel-control",
              "onClick": [Function],
              "onKeyPress": [Function],
              "role": "button",
              "tabIndex": "0",
            },
            "ref": [Function],
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "default/guestbook-app//gbapp-ch/guestbook-app-latest///cassandra-cassandra-service///mysql-wordpress-pd-wordpress-mysql-deployment",
            "nodeType": "host",
            "props": Object {
              "aria-label": "1",
              "children": <React.Fragment>
                <div
                  className="channel-control-start"
                >
                  1
                </div>
              </React.Fragment>,
              "className": "channel-control channel-control-subchannel selected",
              "onClick": [Function],
              "onKeyPress": [Function],
              "role": "button",
              "tabIndex": "0",
            },
            "ref": [Function],
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <div
                  className="channel-control-start"
                >
                  1
                </div>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                  "className": "channel-control-start",
                },
                "ref": null,
                "rendered": 1,
                "type": "div",
              },
              "type": Symbol(react.fragment),
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "default/guestbook-app//gbapp-ch/guestbook-app-latest///mysql-wordpress-pd-wordpress-mysql-service///staging-elasticsearch-elasticsearch-serviceaccount",
            "nodeType": "host",
            "props": Object {
              "aria-label": "2",
              "children": <React.Fragment>
                <div
                  className="channel-control-start"
                >
                  2
                </div>
              </React.Fragment>,
              "className": "channel-control channel-control-subchannel",
              "onClick": [Function],
              "onKeyPress": [Function],
              "role": "button",
              "tabIndex": "0",
            },
            "ref": [Function],
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <div
                  className="channel-control-start"
                >
                  2
                </div>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 2,
                  "className": "channel-control-start",
                },
                "ref": null,
                "rendered": 2,
                "type": "div",
              },
              "type": Symbol(react.fragment),
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`EditorBar components empty 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChannelControl />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`EditorBar components empty 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChannelControl />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
